chia/simulator/full_node_simulator.py

```diff
@@ -49,7 +50,13 @@ async def farm_new_transaction_block(self, request: FarmNewBlockProtocol):
             current_blocks = await self.get_all_full_blocks()
             if len(current_blocks) == 0:
                 genesis = self.bt.get_consecutive_blocks(uint8(1))[0]
-                await self.full_node.blockchain.receive_block(genesis)
+                pre_validation_results: Optional[
+                    List[PreValidationResult]
+                ] = await self.full_node.blockchain.pre_validate_blocks_multiprocessing(
```

Comment:
it seems to me the return type of this function should not be `Optional[...]`

Reply:
What is that? What's wrong with `Optional[List[PreValidationResult]]`?

Reply:
Virtuall all calls I saw to this function was followed by an

```
assert pre_validation_result is not None
```

Which to me suggests that (at least on all of those cases), the function cannot fail (or whatever a `None` return would mean). If the function was just made to never return `None`, we wouldn't need all those asserts. But I didn't look very deeply into what `None` means or under what conditions it's returned, it just looked a bit suspicious to me.

Reply:
I think I may have been looking at tests checking the return value.. sorry about that.